src/Compiler/Scheme/Chez.idr

```diff
@@ -385,49 +390,56 @@ getFgnCall version (n, fc, d) = schFgnDef fc n d version
 
 export
 startChezPreamble : String
-startChezPreamble = unlines
-    [ "#!/bin/sh"
-    , "# " ++ (generatedString "Chez")
-    , ""
-    , "set -e # exit on any error"
-    , ""
-    , "if [ \"$(uname)\" = Darwin ]; then"
-    , "  DIR=$(zsh -c 'printf %s \"$0:A:h\"' \"$0\")"
-    , "else"
-    , "  DIR=$(dirname \"$(readlink -f -- \"$0\")\")"
-    , "fi"
-    , ""  -- so that the preamble ends with a newline
-    ]
+startChezPreamble = """
+  #!/bin/sh
+  # \{ generatedString "Chez" }
+
+  set -e # exit on any error
+
+  if [ "$(uname)" = Darwin ]; then
+    DIR=$(zsh -c 'printf %s "$0:A:h"' "$0")
+  else
+    DIR=$(dirname "$(readlink -f -- "$0")")
+  fi
+
+  """
 
 startChez : String -> String -> String
-startChez appdir target = startChezPreamble ++ unlines
-    [ "export LD_LIBRARY_PATH=\"$DIR/" ++ appdir ++ ":$LD_LIBRARY_PATH\""
-    , "export DYLD_LIBRARY_PATH=\"$DIR/" ++ appdir ++ ":$DYLD_LIBRARY_PATH\""
-    , "export IDRIS2_INC_SRC=\"$DIR/" ++ appdir ++ "\""
-    , "\"$DIR/" ++ target ++ "\" \"$@\""
-    ]
+startChez appdir target = startChezPreamble ++ """
+  export LD_LIBRARY_PATH="$DIR/\{ appdir }":$LD_LIBRARY_PATH
+  export DYLD_LIBRARY_PATH="$DIR/\{ appdir }":$DYLD_LIBRARY_PATH
+  export IDRIS2_INC_SRC="$DIR/\{ appdir }"
+
+  "$DIR/\{ target }" "$@"
+  """
 
 startChezCmd : String -> String -> String -> String -> String
-startChezCmd chez appdir target progType = unlines
-    [ "@echo off"
-    , "set APPDIR=%~dp0"
-    , "set PATH=%APPDIR%" ++ appdir ++ ";%PATH%"
-    , "set IDRIS2_INC_SRC=%APPDIR%" ++ appdir
-    , "\"" ++ chez ++ "\" " ++ progType ++ " \"%APPDIR%" ++ target ++ "\" %*"
-    ]
+startChezCmd chez appdir target progType = """
+  @echo off
+
+  rem \{ generatedString "Chez" }
+
+  set APPDIR=%~dp0
+  set PATH=%APPDIR%\{ appdir };%PATH%
+  set IDRIS2_INC_SRC=%APPDIR%\{ appdir }
+
+  "\{ chez }" \{ progType } "%APPDIR%\{ target }" %*
+  """
 
 startChezWinSh : String -> String -> String -> String -> String
-startChezWinSh chez appdir target progType = unlines
-    [ "#!/bin/sh"
-    , "# " ++ (generatedString "Chez")
-    , ""
-    , "set -e # exit on any error"
-    , ""
-    , "DIR=$(dirname \"$(readlink -f -- \"$0\" || cygpath -a -- \"$0\")\")"
-    , "PATH=\"$DIR/" ++ appdir ++ ":$PATH\""
-    , "export IDRIS2_INC_SRC=\"$DIR/" ++ appdir ++ "\""
-    , "\"" ++ chez ++ "\" " ++ progType ++ " \"$DIR/" ++ target ++ "\" \"$@\""
-    ]
+startChezWinSh chez appdir target progType = """
+  #!/bin/sh
+  # \{ generatedString "Chez" }
+
+  set -e # exit on any error
+
+  DIR=$(dirname "$(readlink -f -- "$0" || cygpath -a -- "$0")")
+  PATH="$DIR/\{ appdir }":$PATH
```

Comment:
```suggestion
  PATH="$DIR/\{ appdir }:$PATH"
```

Reply:
Shouldn't we just assume that the rest of the `$PATH` should be already correctly escaped, and we only care about the path that we prepend?